src/Core/Services/Implementations/UserService.cs

```diff
@@ -1087,13 +1087,24 @@ public async Task<bool> TwoFactorProviderIsEnabledAsync(TwoFactorProviderType pr
             return await CanAccessPremium(user);
         }
 
+        //TODO refactor this to use the below method and enum
         public async Task<string> GenerateEnterprisePortalSignInTokenAsync(User user)
         {
             var token = await GenerateUserTokenAsync(user, Options.Tokens.PasswordResetTokenProvider,
                 "EnterprisePortalTokenSignIn");
             return token;
         }
 
+
+        public async Task<string> GenerateSignInTokenAsync(User user, TokenPurpose purpose)
+        {
+            var token = await GenerateUserTokenAsync(
```

Comment:
Follow the format above, x2-wide parameters per line, inline with the method invocation.
```c#
var token = await GenerateUserTokenAsync(user, Options.Tokens.PasswordResetTokenProvider,
    purpose.ToString());
```

Reply:
done